driver/ppm_tp.c

```diff
@@ -0,0 +1,95 @@
+#include "ppm_tp.h"
+
+const char *tp_names[] = {
+#define X(name, path) path,
+	TP_FIELDS
+#undef X
+};
+
+#ifndef __KERNEL__
+#include <string.h>
+#include "ppm_events_public.h"
+
+typedef struct {
+	ppm_sc_code sc_code;
+	ppm_tp_code tp_code;
+} sc_to_tp_map;
+
+static sc_to_tp_map ppm_sc_to_tp_table[] = {
+	{ PPM_SC_SCHED_PROCESS_EXIT, SCHED_PROC_EXIT},
+	{ PPM_SC_SCHED_SWITCH, SCHED_SWITCH },
+	{ PPM_SC_PAGE_FAULT_USER, PAGE_FAULT_USER },
+	{ PPM_SC_PAGE_FAULT_KERNEL, PAGE_FAULT_KERN },
+	{ PPM_SC_SIGNAL_DELIVER, SIGNAL_DELIVER },
+};
+
+// TODO _Static_assert(sizeof(ppm_sc_to_tp_table) / sizeof(*ppm_sc_to_tp_table) == PPM_SC_TP_LEN, "Wrong number of ppm_sc_to_tp_table entries.");
```

Comment:
I would love to have this in place once again :/

Reply:
This is something we could do in #889 since the syscalls-related ppm_sc and the tracepoints-related ones were detached (ie: the tracepoints based ones were off by 1<<10).

Reply:
Let's say that the danger here is that someone adding a new tracepoints-based event, forgets about adding it here.
Nothing should work though, therefore at the first test, dev should get that something's wrong.